README.md

```diff
@@ -30,160 +30,496 @@ Those tags are generic enough to be used by other biomedical funders
+  --filter-tags TEXT         filter mesh subbranch like disease
+  --test-split FLOAT         split percentage for test data. if None no split.
+  --config PATH              path to config files that defines arguments
+  --help                     Show this message and exit.
+```
+
+### 🔥 Train
+
+Train acts as the entry point command for training all models. You can control
+which model you want to use with an `--approach` flag. This is convenient but 
+also not ideal as different models require different params and considerations.
+This will change in the future and the approach will be a subcommand so each
+model will trigger a different train and have different params.
+
+```
+Usage: grants_tagger train [OPTIONS] [DATA_PATH] [LABEL_BINARIZER_PATH]
+                           [MODEL_PATH]
+
+Arguments:
+  [DATA_PATH]             path to processed JSON data to be used for training
+  [LABEL_BINARIZER_PATH]  path to label binarizer
+  [MODEL_PATH]            path to output model.pkl or dir to save model
+
+Options:
+  --approach TEXT                 tfidf-svm, scibert, cnn, ...  [default:
+                                  tfidf-svm]
+
+  --parameters TEXT               model params in sklearn format e.g.
+                                  {'svm__kernel: linear'}
+
+  --test-data-path PATH           path to processed JSON test data
+  --threshold FLOAT               threshold to assign a tag
+  --data-format TEXT              format that will be used when loading the
+                                  data. One of list,generator  [default: list]
+
+  --test-size FLOAT               float or int indicating either percentage or
+                                  absolute number of test examples  [default:
+                                  0.25]
+
+  --sparse-labels / --no-sparse-labels
+                                  flat about whether labels should be sparse
+                                  when binarized  [default: False]
+
+  --cache-path PATH               path to cache data transformartions
+  --config PATH
+  --help                          Show this message and exit.
+```
+
+Label binarizer converts tags to a Y matrix with one column per tag. This is
+neccesary as this is a multilabel problem so multiple tags may be assigned to
+one grant.
+
+Parameters is a JSON like dump (essentially a stringified dict) with all params
+that the model expects in an sklearn fashion.
+
+### 📈 Evaluate
+
+Evaluate enables evaluation of the performance of various approaches including
+human performance and other systems like MTI, SciSpacy and soon Dimensions. As
+such evaluate has the followin subcommands
+
+#### Model
+
+Model is the generic entrypoint for model evaluation. Similar to train approach
+controls which model will be evaluated. Approach which is a positional argument
+in this command controls which model will be evaluated. Since the data in train
+are sometimes split inside train, the same splitting is performed in evaluate.
```

Comment:
This is something I've faced as well. I ended up saving out the train/test split in `train` and using this in `evaluate`, rather than splitting again with the same seed in `evaluate`. I wonder whether a more elegant pipeline would be to split in a separate function and input the split up data into `train` and `evaluate`? I suppose lots of ML datasets are pre-split, but that doesn't go hand in hand with having the test_size as a parameter to `train`

Reply:
Tricky. I think possibly the best path is to split before but I am still thinking about it :p